spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -38,11 +39,18 @@ private[connector] class BoundStatementBuilder[T](
       val converter = converters(i)
       val columnName = columnNames(i)
       val columnValue = converter.convert(buffer(i))
-      boundStatement.set(columnName, columnValue, CodecRegistryUtil.codecFor(columnTypes(i), columnValue))
-      val serializedValue = boundStatement.getBytesUnsafe(i)
-
-      if (serializedValue != null)
-        bytesCount += serializedValue.remaining()
+      //C* 2.2 (PV4) and Greater Allows us to leave fields Unset
+      if (protocolVersion.toInt < ProtocolVersion.V4.toInt || columnValue != Unset) {
+        if (columnValue == Unset){
+          logWarning(s"""Unset Value set to Null because Current $protocolVersion <
```

Comment:
Please remove log message from there. This is executed for every single value which is inserted and will surely lead to kill the server (log file can grow much faster than the database!). Instead, you can make some flag and report warning once the whole batch is processed. Please do not log anything per column or even per row.